anomalib/models/draem/lightning_model.py

```diff
@@ -30,12 +31,40 @@ class Draem(AnomalyModule):
             be used if left empty.
     """
 
-    def __init__(self, anomaly_source_path: Optional[str] = None):
+    def __init__(
+        self, enable_sspcab: bool = False, sspcab_lambda: float = 0.1, anomaly_source_path: Optional[str] = None
+    ):
         super().__init__()
 
         self.augmenter = Augmenter(anomaly_source_path)
-        self.model = DraemModel()
+        self.model = DraemModel(sspcab=enable_sspcab)
         self.loss = DraemLoss()
+        self.sspcab = enable_sspcab
+
+        if self.sspcab:
+            self.sspcab_activations: Dict = {}
+            self.setup_sspcab()
+            self.sspcab_loss = nn.MSELoss()
+            self.sspcab_lambda = sspcab_lambda
+
+    def setup_sspcab(self):
+        """Prepare the model for the SSPCAB training step by adding forward hooks for the SSPCAB layer activations."""
+
+        def get_activation(name: str) -> Callable:
+            """Retrieves the activations.
+
+            Args:
+                name (str): Identifier for the retrieved activations.
+            """
+
+            def hook(_, __, output: Tensor):
+                """Hook for retrieving the activations."""
+                self.sspcab_activations[name] = output.detach()
```

Comment:
I don't get this. If we detach the outputs then will it be able to compute gradient? My understanding might be wrong here but shouldn't we only detach the input?

Reply:
yes, good catch. I've changed it